src/get30yearWBMGridMET.R

```diff
@@ -1,84 +1,124 @@
 get30yearWBMGridMET <- function(aoi,
-                        path = "data/all",
-                        wb_var = c("runoff","accumswe,AET"),cf = select_cfs$GCM) {
-  
-  # Test vars
-  path <- "data/all"
-  wb_var <- c("runoff")
-  aoi <- park_boundary %>% sf::st_buffer(., dist = .1) %>% sf::st_bbox()
-  
+                                variable = c("runoff"),
+                                cf,
+                                historic_years = c("1981_2010"),
+                                future_years = c("2040_2069", "2070_2099"),
+                                save = TRUE,
+                                path = "data/all",
+                                filename_prefix) {
   # Set lookup vars
-  scenarios <- c('rcp45','rcp85',"historical")
-  rcp <- gsub("^.*\\.","",cf)
-  cfs <- gsub("\\.","_",cf)
-  call_vars <- tidyr::crossing(wb_var,cfs) %>%
-    mutate(rcp = gsub("^.*\\_","",cfs),
-           years = "2040_2069") %>%
-    bind_rows(tidyr::crossing(wb_var,
-                              rcp = "historical",
-                              cfs = "gridmet_historical",
-                              years = "1981_2010"))
-  
-  # Set out dir
-  out_dir <- paste0(path,"/wbm_gridmet_30_year/")
+  scenarios <- c('rcp45', 'rcp85', "historical")
+  rcp <- gsub("^.*\\.", "", cf)
+  cfs <- gsub("\\.", "_", cf)
+  call_vars <- tidyr::crossing(variable, cfs, years = future_years) %>%
+    mutate(rcp = gsub("^.*\\_", "", cfs)) %>%
+    bind_rows(
+      tidyr::crossing(
+        variable,
+        rcp = "historical",
+        cfs = "gridmet_historical",
+        years = historic_years
+      )
+    ) %>% 
+    # fix name for "MIROC-ESM-CHEM" to make sure it matches what WBM link says
+    ## also note, for now since WBM does not have MIROC-ESM for rcp 4.5, this will
+    ## instead use rcp 8.5 if that was a selected model
+    mutate(cfs = if_else(str_detect(cfs, "MIROC-ESM"), "MIROC-ESM-CHEM_rcp85", cfs))
   
-  # Define web call URLs
+  # Define web call URLs, only pulls annual means
   base_call <- "http://screenedcleanedsummaries.s3-website-us-west-2.amazonaws.com/"
-  full_call <- paste0(base_path,call_vars$wb_var,"/",call_vars$rcp,
-                      "/V_1_5_annual_",call_vars$cfs,"_",
-                      call_vars$wb_var,
-                      "_",call_vars$years,"_annual_means_cropped_units_mm.tif")
+  all_calls <- paste0(
+    base_call,
+    call_vars$variable,
+    "/",
+    call_vars$rcp,
+    "/V_1_5_annual_",
+    call_vars$cfs,
+    "_",
+    call_vars$variable,
+    "_",
+    call_vars$years,
+    "_annual_means_cropped_units_mm.tif"
+  )
   
+  #download and crop each individual .tif and save as list
+  tif_list <- purrr::map(all_calls, function(x) {
+    #download a single .tif
+    temp1 <- tempfile()
+    download.file(x, destfile = temp1, method = "curl")
+    
+    # read and crop raster to AOI
+    wb_dat <- terra::rast(temp1) %>%
+      terra::crop(sf::st_transform(aoi, st_crs(wb_dat)))
```

Comment:
on line 65 -- wb_dat in should be changed to "." -> terra::crop(sf::st_transform(aoi, st_crs(.))) 
As wb_dat is not yet a declared variable, it currently errors out here. Otherwise, it works great!!

Reply:
Ah good catch!! Accidentally left that in there from the previous version.